src/test/java/uk/gov/ida/matchingserviceadapter/validator/EidasConditionsValidatorTest.java

```diff
@@ -0,0 +1,109 @@
+import org.opensaml.saml.saml2.core.ProxyRestriction;
+import uk.gov.ida.matchingserviceadapter.exceptions.SamlResponseValidationException;
+import uk.gov.ida.matchingserviceadapter.validators.AssertionTimeRestrictionValidator;
+import uk.gov.ida.matchingserviceadapter.validators.AudienceRestrictionValidator;
+import uk.gov.ida.matchingserviceadapter.validators.EidasConditionsValidator;
+import uk.gov.ida.saml.core.IdaSamlBootstrap;
+
+import java.util.List;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static uk.gov.ida.saml.core.test.builders.AudienceRestrictionBuilder.anAudienceRestriction;
+
+public class EidasConditionsValidatorTest {
+
+    private AssertionTimeRestrictionValidator timeRestrictionValidator;
+    private AudienceRestrictionValidator audienceRestrictionValidator;
+    private Conditions conditions;
+
+    private EidasConditionsValidator validator;
+
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
+
+    @Before
+    public void setUp() {
+        timeRestrictionValidator = mock(AssertionTimeRestrictionValidator.class);
+        audienceRestrictionValidator = mock(AudienceRestrictionValidator.class);
+        conditions = mock(Conditions.class);
+
+        validator = new EidasConditionsValidator(timeRestrictionValidator, audienceRestrictionValidator);
+
+        IdaSamlBootstrap.bootstrap();
+    }
+
+    @Test
+    public void shouldThrowExceptionWhenConditionsIsNull() {
+        expectedException.expect(SamlResponseValidationException.class);
+        expectedException.expectMessage("Conditions is missing from the assertion.");
+
+        validator.validate(null, "any-entity-id");
+    }
+
+    @Test
+    public void shouldThrowExceptionWhenProxyRestrictionElementExists() {
+        expectedException.expect(SamlResponseValidationException.class);
+        expectedException.expectMessage("Conditions should not contain proxy restriction element.");
+
+        when(conditions.getProxyRestriction()).thenReturn(mock(ProxyRestriction.class));
+
+        validator.validate(conditions, "any-entity-id");
+    }
+
+    @Test
+    public void shouldNotThrowExceptionWhenOneTimeUseElementExists() {
+        when(conditions.getOneTimeUse()).thenReturn(mock(OneTimeUse.class));
+        validator.validate(conditions, "any-entity-id");
+    }
+
+    @Test
+    public void shouldValidateNotOnOrAfterIfExists() {
+        DateTime notOnOrAfter = new DateTime();
+        when(conditions.getNotOnOrAfter()).thenReturn(notOnOrAfter);
+
+        validator.validate(conditions, "any-entity-id");
+
+        verify(timeRestrictionValidator).validateNotOnOrAfter(notOnOrAfter);
+    }
+
+    @Test
+    public void shouldNotValidateNotOnOrAfterIfExists() {
+        DateTime notOnOrAfter = null;
+        when(conditions.getNotOnOrAfter()).thenReturn(notOnOrAfter);
+
+        validator.validate(conditions, "any-entity-id");
```

Comment:
Is this actually testing that its not being validated?

Reply:
See the above comment; I'm not really sure, I just used what already existed.